swift/send-message/message-types/discord/Sources/send-discord.swift

```
@@ -0,0 +1,113 @@
+
+        let bodyResponse: ByteBuffer = try await response.body.collect(upTo: 1024 * 1024) // 1 MB
+        let jsonResponse = try JSONSerialization.jsonObject(with: bodyResponse.getData(at: 0, length: bodyResponse.readableBytes)!, options: []) as! Array<[String: Any]>            
+
+        //The response contains all users with the queried username as a substring in their username or nickname.
+        //The below loop makes sure the currect user is messaged.
+        for user  in jsonResponse{
+            let currentUsername = ((user["user"]) as! [String: Any])["username"] as! String
+            
+            if currentUsername == username{
+                return ((user["user"]) as! [String: Any])["id"] as! String
+            }      
+        }
+        throw MessengerError.misconfigurationError(error: "No users with username \(username) found in server")
+    }
+
+    //Returns dm channel ID. If there isn't already a dm channel with a user one is made.
+    private func createDM(recipient_id: String) async throws -> String{
+        let targetURL = "https://discord.com/api/v10/users/@me/channels"
+        var request = HTTPClientRequest(url: targetURL)
+        let jsonRecipientID: [String: String] = ["recipient_id": "\(recipient_id)"]
+
+        request.method = .POST
+        request.headers.add(name: "Content-Type", value: "application/json")
+        request.headers.add(name: "Authorization", value: botToken)
+        request.body = .bytes(ByteBuffer(data: (try JSONSerialization.data(withJSONObject: jsonRecipientID))))
+        
+        let response:HTTPClientResponse
+        do{
+            response = try await httpClient.execute(request, timeout: .seconds(30))
+        }catch{
+            throw MessengerError.providerError(error: "Request url not responding or connection timed out")
+        }
+        if response.status != .ok {
+            throw MessengerError.validationError(error: "Unable to create dm, API Status Code: \(response.status)")
```

make sure you're not swallowing a discord error here